library/Requests/Transport/cURL.php

```diff
@@ -546,4 +561,23 @@ public static function test($capabilities = array()) {
 
 		return true;
 	}
+
+	/**
+	 * Get the correct "Expect" header for the given request data.
+	 *
+	 * @param string|array $data Data to send either as the POST body, or as parameters in the URL for a GET/HEAD
+	 * @return string The "Expect" header
+	 */
+	protected function get_expect_header($data) {
+		if (is_array($data)) {
+			$iterator = new RecursiveIteratorIterator(new RecursiveArrayIterator($data));
+			$data     = '';
+
+			foreach ($iterator as $datum) {
+				$data .= $datum;
+			}
+		}
+
+		return strlen($data) >= 1048576 ? '100-Continue' : '';
```

Comment:
This is not very efficient given we're only interested in the byte size of the array. The concatenation of the string can be expensive in terms of memory operations, and then you still haven't counted the bytes yet.

How about this instead:
```suggestion
		if (is_array($data)) {
			$iterator = new RecursiveIteratorIterator(new RecursiveArrayIterator($data));
			$bytesize = 0;

			foreach ($iterator as $datum) {
				$bytesize += strlen((string)$datum);
			}
		} else {
			$bytesize = strlen((string)$data);
		}

		return $bytesize >= 1048576 ? '100-Continue' : '';
```